test/e2e/tests/ppom/security-alerts-api/security-alerts-api-contract-interaction.spec.ts

```diff
@@ -0,0 +1,316 @@
+import { MockttpServer } from 'mockttp';
+import { SECURITY_ALERTS_PROD_API_BASE_URL } from '../constants';
+
+import FixtureBuilder from '../../../fixture-builder';
+
+import { defaultGanacheOptions, withFixtures } from '../../../helpers';
+import { SecurityAlertResponse } from '../../../../../app/scripts/lib/ppom/types';
+import {
+  initializePages,
+  openDappAndClickButton,
+  openDappAndTriggerSignature,
+  SignatureType,
+} from '../../confirmations/signatures/signature-helpers';
+import Confirmation from '../../../page-objects/pages/confirmations/redesign/confirmation';
+
+const SELECTED_ADDRESS = '0x5cfe73b6021e818b776b421b1c4db2474086a7e1';
+
+const CONTRACT_ADDRESS = {
+  BalanceChecker: '0xb1f8e55c7f64d203c1400b9d8555d050f94adf39',
+  BUSD: '0x4fabb145d64652a948d72533023f6e7a623c7c53',
+  BUSDImplementation: '0x2a3f1a37c04f82aa274f5353834b2d002db91015',
+  OffChainOracle: '0x52cbe0f49ccdd4dc6e9c13bab024eabd2842045b',
+  BAYC: '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d',
+  ENSRegistryWithFallback: '0x00000000000c2e074ec69a0dfb2997ba6c7d2e1e',
+  USDC: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+} as const;
+
+const preferencesOptions = {
+  securityAlertsEnabled: true,
+  preferences: {
+    redesignedTransactionsEnabled: true,
+    redesignedConfirmationsEnabled: true,
+    isRedesignedConfirmationsDeveloperEnabled: true,
+  },
+};
+
+const expectedTitle = 'This is a deceptive request';
+
+describe('Security Alerts API - Contract Interaction @no-mmi', function () {
+  describe('Malicious Contract interaction', function () {
+    const maliciousTransferAlert: SecurityAlertResponse = {
+      block: 1,
+      result_type: 'Malicious',
+      reason: 'raw_native_token_transfer',
+      description:
+        'Interaction with a known malicious address: 0x00008f1149168c1d2fa1eba1ad3e9cd644510000',
+      features: ['Interaction with a known malicious address'],
+    };
+
+    async function mockRequest(
+      server: MockttpServer,
+      response: SecurityAlertResponse,
+    ): Promise<void> {
+      await server
+        .forPost(`${SECURITY_ALERTS_PROD_API_BASE_URL}/validate/0x1`)
+        .withJsonBodyIncluding({
+          method: 'eth_sendTransaction',
+          params: [
+            {
+              from: SELECTED_ADDRESS,
+              data: '0xef5cfb8c0000000000000000000000000b3e87a076ac4b0d1975f0f232444af6deb96c59',
+              to: '0x00008f1149168c1d2fa1eba1ad3e9cd644510000',
+              value: '0x0',
+            },
+          ],
+        })
+        .thenJson(201, response);
+    }
```

Comment:
It should be possible to create a common function to avoid code repetition with method `mockRequest `.